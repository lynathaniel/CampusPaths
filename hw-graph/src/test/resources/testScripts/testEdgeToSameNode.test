# Tests for the handling of duplicates in graphs.

# Create a graph
CreateGraph graph1
CreateGraph graph2

# Add nodes to graph
AddNode graph1 n1
AddNode graph1 n2

AddNode graph2 n3
AddNode graph2 n4

# Connect node to self using edge
AddEdge graph1 n1 n1 e1
AddEdge graph1 n2 n2 e2

AddEdge graph2 n3 n3 e3
AddEdge graph2 n4 n4 e4

# Print out the children of the outgoing edges from n1, n2, n3 & n4
ListChildren graph1 n1
ListChildren graph1 n2

ListChildren graph2 n3
ListChildren graph2 n4