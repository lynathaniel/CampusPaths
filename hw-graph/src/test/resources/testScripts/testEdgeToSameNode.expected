# Tests for the handling of duplicates in graphs.

# Create a graph
created graph graph1
created graph graph2

# Add nodes to graph
added node n1 to graph1
added node n2 to graph1

added node n3 to graph2
added node n4 to graph2

# Connect node to self using edge
added edge e1 from n1 to n1 in graph1
added edge e2 from n2 to n2 in graph1

added edge e3 from n3 to n3 in graph2
added edge e4 from n4 to n4 in graph2

# Print out the children of the outgoing edges from n1, n2, n3 & n4
the children of n1 in graph1 are: n1(e1)
the children of n2 in graph1 are: n2(e2)

the children of n3 in graph2 are: n3(e3)
the children of n4 in graph2 are: n4(e4)